dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/longrunning/LongRunningTask.java

```diff
@@ -0,0 +1,53 @@
+package io.openliberty.microprofile.telemetry.internal_fat.apps.longrunning;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Test;
+
+import componenttest.app.FATServlet;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.servlet.annotation.WebServlet;
+
+@WebServlet("/longrunning")
+@ApplicationScoped
+public class LongRunningTask extends FATServlet {
+
+    @Inject
+    Tracer tracer;
+
+    @Inject
+    OpenTelemetry openTelemetry;
+
+    private static final String INVALID_SPAN_ID = "0000000000000000";
+
+    public static final String SUCCESS_MESSAGE = "COMPLETED SUCCESSFULLY";
+
+    @Test
+    public void testLongRunningTask() {
+        Span span = tracer.spanBuilder("longRunningTask").startSpan();
+        try {
+            span.makeCurrent();
```

Comment:
```suggestion
        try (Scope scope = span.makeCurrent()) {
```

Need to do the `makeCurrent()` in a try-with-resources like this to make sure it gets made non-current, otherwise the context gets left on the thread.

We always have to ensure we end the span and the scope - the span so that it finishes and is exported, the scope so that we don't leave the wrong context hanging around.